python/tank/deploy/tank_commands/unregister_folders.py

```diff
@@ -0,0 +1,311 @@
+                
+                log.info("This command is used to unregister folders that have been created via the Toolkit "
+                         "folder creation. When applications are launched and folders are created, new entries "
+                         "are added to the Shotgun FilesystemLocation table. These are then used to track the "
+                         "relationship between Shotgun objects and folders on disk. If you ever need to "
+                         "undo these associations, you can use this command. ")
+                log.info("")
+                log.info("You can use it by passing a Shotgun object, e.g.")
+                log.info("> tank Shot ABC123 unregister_folders")
+                log.info("")
+                log.info("Alternatively, you can use it by passing one or more paths, e.g.")
+                log.info("> tank unregister_folders /path/to/folder_a /path/to/folder_b")
+                log.info("")
+                
+            else:
+                paths = args
+                log.info("Will process the following folders:")
+                for p in paths:
+                    log.info(" - %s" % p)
+                log.info("")
+                self._unregister_paths(paths, log, prompt=True)
+        
+        
+    def run_noninteractive(self, log, parameters):
+        """
+        API accessor
+        
+        :param log: Std logging object
+        :param parameters: Std tank command parameters dict
+        :returns: List of dictionaries to represents the items that were unregistered.
+                  Each dictionary has keys path and sg_id.
+                  Note that the shotgun ids returned will refer to retired objects in 
+                  Shotgun rather than live ones.
+        """
+        # validate params and seed default values
+        computed_params = self._validate_parameters(parameters) 
+        
+        # must have a path or an entity
+        if computed_params["path"] is None and computed_params["entity"] is None:
+            raise TankError("Must either specify a path or an entity!")
+        # ... but not both
+        if computed_params["path"] and computed_params["entity"]:
+            raise TankError("Cannot specify both a path and an entity!")
+            
+        if computed_params["path"]:
+            path = computed_params["path"]
+            return self._unregister_paths([path], log, prompt=False)
+        
+        if computed_params["entity"]:
+            entity = computed_params["entity"]
+            if "id" not in entity:
+                raise TankError("Entity dictionary does not contain an id key!")
+            if "type" not in entity:
+                raise TankError("Entity dictionary does not contain a type key!")
+            return self._unregister_entity(entity, log, prompt=False)
+            
+
+    def _unregister_paths(self, paths, log, prompt):
+        """
+        Unregisters a path.
+        
```

Comment:
we don't have docstrings for any of the tank commands - they all follow the same conventions. But i'll add some :) Good suggestion.